Seminars/Sem.04/Pract.04/Task03.cpp

```diff
@@ -0,0 +1,267 @@
+#include <iostream>
+#include <fstream>
+
+const int MAX_LENGTH = 25;
+const int MAX_NO_OF_GAMES = 25;
+const int BUFF = 1024;
+class Game {
+private:
+	char title[MAX_LENGTH];
+	float price;
+	bool isAvailable;
+public:
+	// Default constructor
+	Game() : price(0.0), isAvailable(false){ title[0] = '\0'; }
+	// Parameterized constructor
+	Game(const char title[], float price, bool isAvailable) {
+		setTitle(title);
+		setPrice(price);
+		setAvailability(isAvailable);
+
+	}
+	const char* getTitle()const {
+		return title;
+	}
+	float getPrice()const {
+		
+			return price;
+	
+	}
+	bool getAvailability()const {
+			return isAvailable;
+	
+	}
+	void setTitle(const char title[]) {
+		int index = 0;
+		while (title[index] != '\0') {
+			this->title[index] = title[index];
+			index++;
+		}
+		this->title[index] = '\0';
+	}
+	void setPrice(float price)
+	{
+		if (price >= 0) {
+			this->price = price;
+		}
+	}
+    void setAvailability(bool isAvailable) {
+
+		this->isAvailable = isAvailable;
+
+		}
+	
+	bool isFree() const { return price == 0.0; }
+	
+
+	void printInformation()const {
+
+		std::cout << "Title:" << getTitle() << std::endl;
+		std::cout << "Price:" << getPrice() << std::endl;
+		std::cout << "Availability: " << (isAvailable ? "Available" : "Not available") << std::endl;
+
+	}
+
+};
+
+class GamePlatform {
+private:
+	Game games[MAX_NO_OF_GAMES];
+	int numOfGames;
+public:
+	GamePlatform():numOfGames(0){}
+
+
+	void addGame( const Game&game) {
+		if (numOfGames < MAX_NO_OF_GAMES) {
+			games[numOfGames] = game;
+			numOfGames++;
+			std::cout << "Game added successfully!" << std::endl;
+		}
+		else {
+			std::cout << "Library is full, cannot add game." << std::endl;
+		}
+	}
+
+	void DisplayGameByIndex(const int index)const {
+		if (index >= 0 && index < numOfGames) {
+			const Game& game = games[index];
```

Comment:
The referene here is useless, you can just write games[index].printInformation()